src/main/java/com/mongodb/jdbc/MongoJsonSchema.java

```diff
@@ -0,0 +1,67 @@
+package com.mongodb.jdbc;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang3.builder.ToStringBuilder;
+import org.apache.commons.lang3.builder.ToStringStyle;
+
+public class MongoJsonSchema {
+    public String bsonType;
+    public Map<String, MongoJsonSchema> properties;
+    public Set<MongoJsonSchema> anyOf;
+    public Set<String> required;
+    public MongoJsonSchema items;
+    public boolean additionalProperties;
+
+    public MongoJsonSchema() {}
+
+    public static MongoJsonSchema createEmptyObjectSchema() {
+        MongoJsonSchema ret = new MongoJsonSchema();
+        ret.bsonType = "object";
+        ret.properties = new HashMap<String, MongoJsonSchema>();
+        ret.required = new HashSet<String>();
+        return ret;
+    }
+
+    public static MongoJsonSchema createScalarSchema(String type) {
+        MongoJsonSchema ret = new MongoJsonSchema();
+        ret.bsonType = type;
+        return ret;
+    }
+
+    @SafeVarargs
```

Comment:
This will make the creation of `MongoJsonSchema` in the `MongoSQLDatabaseMetaData` methods much cleaner.